components/SearchBox/components/SearchBox.tsx

```diff
@@ -0,0 +1,199 @@
+import {
+  MagnifyingGlassIcon,
+  ChevronLeftIcon,
+} from '@heroicons/react/24/outline';
+import type { Results, Nullable } from '@orama/orama';
+import clx from 'classnames';
+import { useRouter } from 'next/navigation';
+import { useState, useRef, type FC, useEffect } from 'react';
+
+import styles from '@/components/SearchBox/components/index.module.css';
+import { PoweredBy } from '@/components/SearchBox/components/PoweredBy';
+import { SearchResult } from '@/components/SearchBox/components/SearchResult';
+import { SeeAll } from '@/components/SearchBox/components/SeeAll';
+import { orama, getInitialFacets } from '@/components/SearchBox/lib/orama';
+import { useClickOutside } from '@/components/SearchBox/lib/useClickOutside';
+
+import { EmptyState } from './EmptyState';
+import { NoResults } from './NoResults';
+
+export type SearchDoc = {
+  id: string;
+  path: string;
+  pageTitle: string;
+  siteSection: string;
+  pageSectionTitle: string;
+  pageSectionContent: string;
+};
+
+type SearchResults = Nullable<Results<SearchDoc>>;
+
+type SearchBoxProps = {
+  onClose: () => void;
+};
+
+export const SearchBox: FC<SearchBoxProps> = props => {
+  const [searchTerm, setSearchTerm] = useState('');
+  const [searchResults, setSearchResults] = useState<SearchResults>(null);
+  const [selectedFacet, setSelectedFacet] = useState<number>(0);
+  const [searchError, setSearchError] = useState<Nullable<Error>>(null);
+
+  const router = useRouter();
+  const searchInputRef = useRef<HTMLInputElement>(null);
+  const searchBoxRef = useRef<HTMLDivElement>(null);
+
+  useClickOutside(searchBoxRef, () => {
+    reset();
+    props.onClose();
+  });
+
+  useEffect(() => {
+    searchInputRef.current?.focus();
+    getInitialFacets().then(setSearchResults).catch(setSearchError);
+
+    return () => {
+      reset();
+    };
+  }, []);
+
+  useEffect(() => {
+    search(searchTerm);
+  }, [searchTerm, selectedFacet]);
+
+  function search(term: string) {
+    orama
+      .search({
+        term,
+        limit: 8,
```

Comment:
These default values should be outsourced to a constant on next.constants.mjs such as `DEFAULT_ORAMA_QUERY_PARAMS`

Reply:
With comment pointing to orama docs